docs/core-design.md

```diff
@@ -0,0 +1,354 @@
+# Multi-Network design
+
+## Goal
+
+- Define the network handler for Kubernetes clusters  
+- Define behaviours for the handler  
+- Define integration and interactions with [Dynamic Resource Allocation](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation) (DRA)  
+- Standardize APIs for driver implementations
+
+## Non-Goal
+
+Provide implementation for the driver. See [https://github.com/kubernetes-sigs/cni-dra-driver](https://github.com/kubernetes-sigs/cni-dra-driver) for an example implementation.
+
+## Terminology
+
+**Default PodNetwork** \- This is the initial cluster-wide PodNetwork provided during cluster creation that is available to the Pod when no additional networking configuration is provided in Pod spec.  
+**Primary PodNetwork** \- This is the PodNetwork inside the Pod’s network namespace whose interface is used for the default gateway (0.0.0.0) in the default virtual routing and forwarding table (VRF).  
+**DRA** \- Dynamic Resource Allocation is an API allowing device sharing across multiple workloads (see the official [page](https://kubernetes.io/docs/concepts/scheduling-eviction/dynamic-resource-allocation/) for more details)
+
+## Background
+
+### API and Implementation
+
+This design focuses mainly on the user-facing APIs definitions and how these are going to interact with DRA.  
+The Implementation component is the DRA driver delivered by the provider that handles the network interface for the pod, based on the DRA configuration. It also integrates with the DRA scheduling mechanism based on the implementation specific logic. This area is fully owned by the provider for the Multi-Network capability. This proposal will try to define standardization for these implementations, but not the implementation itself.
```

Comment:
you may consider provide some e2e test that standardize these behaviors like gateway and kubernetes does, otherwise if the APIs are the same but the semantics are different you will not achieve much for users .... users want portability to be able to depend on the API and behaviors and avoid locking in a specific solution

Reply:
Definitely the plan, let me add that to the text.